File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.cpp
class Solution
{
public:
    double findMaxAverage(vector<int> &nums, int k)
    {
        double v = 0;
        for (int i = 0; i < k; ++i)
        {
            v += nums[i];
        }

        double maxSum = v;
        int i = 0, j = k, l = nums.size();
        while (j < l)
        {
            v = (v - nums[i]) + nums[j];
            maxSum = max(maxSum, v);
            ++i;
            ++j;
        }

        return maxSum / k;
    }
};

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        temp = sum(nums[0:k])
        answer = temp

        for i in range(k, len(nums)):
            temp -= nums[i - k]
            temp += nums[i]
            answer = max(answer, temp)

        return answer / k
